components/mediation/tasks/org.wso2.micro.integrator.ntask.core/src/main/java/org/wso2/micro/integrator/ntask/core/internal/TasksDSComponent.java

```diff
@@ -82,12 +86,17 @@ protected void activate(ComponentContext ctx) {
             if (getTaskService() == null) {
                 taskService = new TaskServiceImpl();
             }
+
+            ClusterCoordinator clusterCoordinator = new ClusterCoordinator(dataSourceService);
+            clusterCoordinator.startCoordinator();
```

Comment:
We have to retrieve  CarbonDataSource dataSource = dataSourceService.getDataSource(RDBMSConstantUtils.COORDINATION_DB_NAME); and do this is only if it is valid.